### PR TITLE
gh-103583: Add codecs and maps to _codecs_* module state

### PR DESCRIPTION
**Pull request by erlend-aasland**
- gh-103092
- gh-103583

### CODE REVIEW

**Comment by erlend-aasland** [comment]
Also: this is _very much WIP_. With this PR, I've tried to keep the diff small, but I really do not like all these macro templates. It all feels very fragile to me.

**Comment by erlend-aasland** [comment]
Most of the remaining globals are just static data, as far as I can see. Perhaps the solution is as easy as making sure they all live in the const data segment and forget about module state.

**Comment by erlend-aasland** [comment]
@corona10, as the code owner for these modules: what do you think about the approach taken in this PR? IMO, I'm not too fond of using macros for so much templating, but I went for it nonetheless since that's the approach already taken by this part of the code base.

**Comment by corona10** [comment]
@erlend-aasland Will take a look by this weekend :) Thanks

File: Modules/cjkcodecs/cjkcodecs.h
Comment (corona10) [inline Modules/cjkcodecs/cjkcodecs.h:303]: Please check that if PyObject_CallOneArg returns NULL :)

File: Modules/cjkcodecs/cjkcodecs.h
Comment (corona10) [inline Modules/cjkcodecs/cjkcodecs.h:267]: Wow, readability will be frustrating (it's not your fault), I will think about a better way.. :(

File: Modules/cjkcodecs/cjkcodecs.h
Comment (erlend-aasland) [reply Modules/cjkcodecs/cjkcodecs.h:303]: `res` is propagated to the caller ;)

File: Modules/cjkcodecs/cjkcodecs.h
Comment (erlend-aasland) [reply Modules/cjkcodecs/cjkcodecs.h:267]: I know, readability is already frustrating with these modules. If we are to keep the diff down, we just have to go with the macro template style already used. An alternative is to refactor the modules first, but that bears a risk.

**Comment by erlend-aasland** [comment]
Are you happy with these changes as they stand, @corona10?

**Review by corona10 — APPROVED** [review]
lgtm :)

File: Modules/cjkcodecs/_codecs_iso2022.c
Comment (erlend-aasland) [inline Modules/cjkcodecs/_codecs_iso2022.c]: I'd like to make one more change before landing. This `idx` manipulation is quite ugly. I would prefer to at least to hide it in a macro before landing this.

Comment (erlend-aasland) [reply Modules/cjkcodecs/_codecs_iso2022.c]: I hid the ugly code in 65af11088e2d25b359ad81e88df8e6a5c668c969.

**Comment by erlend-aasland** [comment]
Thanks for the review, Dong-hee! :)

**Pull request merged**
